### ISSUE TITLE
Webpack scope hoisting changes System, Component names

### ISSUE THREAD

**Issue by canadaduane** [gohyperr/hecs#31]
When I write a component such as the following, Webpack's scope hoisting changes the Component's name:
```
import { Component, BooleanType } from "hecs";
export class PointerPlane extends Component {
...
}
```

So `PointerPlane` becomes `PointerPlane_PointerPlane` in this example.

I need component names to remain consistent across networks and code bases. Perhaps we should require a `static name` in each component and system, instead of relying on `this.constructor.name`?

Related:
https://github.com/webpack/webpack/issues/8132

**Comment by ashconnell** [comment]
Agreed. The plugins in this repo have a webpack config to keep class names, but that also means you need to set this up in your own project if you use custom components and systems.

**Comment by ashconnell** [comment]
If you're also using webpack you can use this config https://github.com/gohyperr/hecs/blob/master/packages/hecs-plugin-core/webpack.config.js#L24-L30

**Comment by canadaduane** [comment]
I'm using Webpack, but I can't use Terser for some reason (the resulting code throws an error, which I have not had time to diagnose).

When you say "Agreed" are you saying a `static name` is the right way to move forward?

**Comment by ashconnell** [comment]
I agreed it was a problem. The reason I don't particularly like the `static name` approach is that it requires you to name the component twice. It's cumbersome and what happens if you name them differently?

**Comment by canadaduane** [comment]
I tried assigning `static name = 'Modell'` to the Model component in tests:
```
class Model extends Component {
  static name = 'Modell'
}
```

but got this:

```
 FAIL  tests/components.test.js
  ● Test suite failed to run

    TypeError: Cannot assign to read only property 'name' of function 'function Model() {
        (0, _classCallCheck2["default"])(this, Model);
        return _super.apply(this, arguments);
      }'
```

So if we do go with a static property, it would have to be something other than 'name' :(

**Comment by canadaduane** [comment]
I've looked in to this a bit more. The webpack setting that is causing the change of names is called Module Concatenation, and as of Webpack 4.0 is built-in:

https://webpack.js.org/plugins/module-concatenation-plugin/

The "solution" for now is to disable this enabled-by-default setting:

```
  optimization: {
    concatenateModules: false,
  },
```

Unfortunately, this has the side-effect of slowing down production load times:

"[Module Concatenation] concatenate[s] the scope of all your modules into one closure and allow for your code to have a faster execution time in the browser"

**Comment by canadaduane** [comment]
If you're considering a switch to Typescript, this looks like a fairly elegant solution:

```
@name('RigidBody')
export class RigidBody extends Component {
  // ...
}
```

https://stackoverflow.com/a/62471619/159344

**Comment by canadaduane** [comment]
I realized today when I refactored some code that it's important to decouple the class name from what happens on the network / in save-game files: for example, I changed a component from "FixedJoint" to "BallJoint" and if I'd have had save-game data, I would have needed to find a way to keep the new `BallJoint` class handling what is recorded as a "FixedJoint" component in serialized file data.

So while I agree it would be "nice" to have the class name correspond with the es6 class name, I think these other concerns (down-line minification issues, decoupling of network/file serialization from the names of classes) outweigh the benefits of pure simplicity.
